app/routes/($locale).products.$handle.jsx

```diff
@@ -84,33 +91,62 @@ function redirectToFirstVariant({ product, request }) {
   );
 }
 
+export async function clientLoader({serverLoader}) {
+  const serverData = await serverLoader();
+  const clientData = await new Promise(async (resolve) => {
+    nostojs(async (api) => {
+      const result = await api
+        .defaultSession()
+        .viewProduct({
+          product_id: serverData.product?.id,
+          selected_sku_id: serverData.product?.selectedVariant?.sku,
+        })
+        .setPlacements(['productpage-nosto-1'])
```

Comment:
Here, the placements should be hard-coded. Otherwise, the recommendation rendering may not work properly